doc/design/chunk_header.md

```diff
@@ -1,18 +1,31 @@
+Chunks are the transport capsules in iceoryx. They store data from a publisher as payload and are sent to one or more subscriber. Furthermore, there is some meta information which is stored alongside the payload, e.g. the size and the origin of the chunk. This data is composed in the `ChunkHeader` and located at the front of the chunk. Custom meta information can be added to extend the data from the `ChunkHeader` and tailor the chunk to specific use cases. While this makes the chunk layout more complex, this complexity would otherwise be distributed at different locations all over the code base, e.g. in the request/response feature. Additionally, the adjustments for the user-header makes arbitrary alignments of the user-payload trivial to implement.
 
 ## Terminology
 
 | Name              | Description                                              |
 | :---------------- | :------------------------------------------------------- |
 | Chunk             | a piece of memory                                        |
-| Chunk Header      | contains meta information related to the chunk           |
-| Custom Header     | contains custom meta information, e.g. timestamps        |
-| Payload           | the user data                                            |
-| Back-Offset       | offset stored in front of the payload to calculate back to the chunk header (for the most simple case it will overlap with the payload offset stored in the Chunk Header) |
+| Chunk-Header      | contains meta information related to the chunk           |
+| Chunk-Payload     | the part of the chunk for the payload which is split into user-header and user-payload |
+| User-Header       | contains custom meta information, e.g. timestamps        |
+| User-Payload      | the user data with custom alignment                      |
+| Back-Offset       | offset stored in front of the user-payload to calculate back to the chunk-header (for the most simple case it will overlap with the user-payload offset stored in the Chunk-Header) |
+
+Framing with terminology
+```
++=============================================================================+
+|                                 Chunk                                       |
++===================+=========================================================+
+|                   |                    Chunk-Payload                        |
+|   Chunk-Header    +===============+=======+====================+============+
+|                   |  User-Header  |   ¦ 🢑 |    User-Payload    |  Padding   |
```

Comment:
This is intentional to show that the back-offset will be put in front of the user-payload, that it's dependent on the user payload and that it can overlap with the `userPayloadOffset` in `ChunkHeader`, which is shown in a diagram further down. Therefore a `¦` instead of a `|`